File: MarcenariaDotNet/Dtos/Geometrias/Cubo.cs
using System.Text.Json.Serialization;
using MarcenariaDotNet.Utils;

namespace MarcenariaDotNet.Dtos.Geometrias;

public struct Cubo : Geometria
{
  public String Estrutura { get; init; }
  [JsonConverter(typeof(CentimeterConverter))]
  public double Lado { get; init; }

    double Geometria.GetArea()
    {
        return Math.Sqrt(3) * Lado / 2;
    }

    string Geometria.GetEstrutura()
    {
        return Estrutura;
    }

    string Geometria.GetGeometria()
    {
       return "cubo";
    }
}

File: codes/java/marcenaria-java-activej/src/main/java/github/gmess/models/geometrias/Cubo.java
package github.gmess.models.geometrias;

import github.gmess.utils.NumberUtils;

/**
 * Cubo
 */
public record Cubo(
  String geometria,
  String estrutura,
  String lado
) implements Geometria {
  
  @Override
  public final String getGeometria() {
    return "cubo";
  }

  @Override
  public final String getEstrutura() {
    return this.estrutura;
  }

  @Override
  public final double getArea() {
    final double lado = NumberUtils.centimetersToDouble(this.lado);
    return 6 * Math.pow(lado, 2);
  }

}
